services/worker/tests/job_runners/split/test_descriptive_statistics.py

```diff
@@ -471,6 +474,51 @@ def descriptive_statistics_string_text_partial_expected(datasets: Mapping[str, D
     return {"num_examples": df.shape[0], "statistics": expected_statistics, "partial": True}
 
 
+@pytest.fixture
+def audio_statistics_expected() -> dict:  # type: ignore
+    audio_lengths = [1.0, 2.0, 3.0, 4.0]  # datasets consists of 4 audio files of 1, 2, 3, 4 seconds lengths
+    audio_statistics = count_expected_statistics_for_numerical_column(
+        column=pd.Series(audio_lengths), dtype=ColumnType.FLOAT
+    )
+    expected_statistics = {
+        "column_name": "audio",
+        "column_type": ColumnType.AUDIO,
+        "column_statistics": audio_statistics,
+    }
+    nan_audio_lengths = [1.0, 3.0]  # take first and third audio file for this testcase
+    nan_audio_statistics = count_expected_statistics_for_numerical_column(
+        column=pd.Series(nan_audio_lengths), dtype=ColumnType.FLOAT
+    )
+    expected_nan_statistics = {
+        "column_name": "audio_nan",
+        "column_type": ColumnType.AUDIO,
+        "column_statistics": nan_audio_statistics,
```

Comment:
does it include `"nan_count": 2`? From the code it seems not. But the array contains two audio files and two NaN, right?

Reply:
oh that was a bug than you! fixed [4f7db83](https://github.com/huggingface/datasets-server/pull/2612/commits/4f7db83f772f750f422a44c73ae7e1da8e1fefc8)